components/Recommendations/Recommendations.tsx

```diff
@@ -156,6 +158,7 @@ export default function Recommendations({
       return (
         <Recommendation
           address={item}
+          navigation={navigation}
```

Comment:
We should never pass navigation IMO. And use `useRouter` instead

Reply:
done